### DISCUSSION TITLE
Gradients w.r.t. coordinates in ktraj

### DISCUSSION THREAD

**Discussion by tobitk** [mmuckley/torchkbnufft#96] [Q&A] [answered]
Hi,
thanks for the great package! 
Is it right that both nufft and adjoint nufft are only differentiable w.r.t. the input image/k-space but not w.r.t. the coordinates in the input trajectory?
Is it feasible to add gradients w.r.t. to the coordinates as well?

Best,
Tobit

**Comment by mmuckley** [comment, marked as answer]
Migrating to a Discussion since this does not document a bug.

Hello @tobitk, you are correct. You might want to look into methods like [BJORK](https://github.com/guanhuaw/Bjork) for what you want to do.

CC @guanhuaw

**Reply by tobitk** [reply]
Thanks for you answer! In the end, [this repo](https://github.com/albangossard/Bindings-NUFFT-pytorch) solved my problem.
